config/interpolate_funcs.go

```diff
@@ -292,7 +297,7 @@ func interpolationFuncElement() ast.Function {
 					"invalid number for index, got %s", args[1])
 			}
 
-			v := list[index%len(list)]
+			v := list.Element(index)
```

Comment:
Easiest to just push this down to the StringList type - with the mod behavior living down there.